libs/cardano-ledger-api/test/Test/Cardano/Ledger/Api/State/Imp/QuerySpec.hs

```diff
@@ -0,0 +1,314 @@
+  ConwayGovState,
+  EraGov (..),
+  GovAction (..),
+  GovActionPurpose (..),
+  PrevGovActionId (..),
+  Voter (DRepVoter),
+  cgEnactStateL,
+  ensCommitteeL,
+ )
+import Cardano.Ledger.Conway.PParams (
+  dvtCommitteeNoConfidence,
+  dvtCommitteeNormal,
+  dvtUpdateToConstitution,
+  ppCommitteeMaxTermLengthL,
+  ppDRepVotingThresholdsL,
+  ppGovActionDepositL,
+  ppGovActionLifetimeL,
+ )
+import Cardano.Ledger.Core
+import Cardano.Ledger.Credential (Credential (KeyHashObj))
+import Cardano.Ledger.Keys (
+  KeyHash,
+  KeyRole (..),
+ )
+import Cardano.Ledger.Shelley.LedgerState
+import Data.Default.Class (Default (..))
+import Data.Foldable (Foldable (..))
+import qualified Data.Map.Strict as Map
+import qualified Data.Set as Set
+import Lens.Micro ((&), (.~))
+import Lens.Micro.Mtl
+import Test.Cardano.Ledger.Conway.ImpTest
+import Test.Cardano.Ledger.Core.Rational (IsRatio (..))
+import Test.Cardano.Ledger.Imp.Common
+
+spec ::
+  forall era.
+  ( HasCallStack
+  , ConwayEraImp era
+  , GovState era ~ ConwayGovState era
+  ) =>
+  SpecWith (ImpTestState era)
+spec =
+  it "Committee queries" $ do
+    setPParams
+    drep <- setupSingleDRep
+    c1 <- freshKeyHash
+    c2 <- freshKeyHash
+    c3 <- freshKeyHash
+    c4 <- freshKeyHash
+    c5 <- freshKeyHash
+    c6 <- freshKeyHash
+    c7 <- freshKeyHash
+    c8 <- freshKeyHash
+    ga1 <-
+      electCommittee
+        SNothing
+        drep
+        Set.empty
+        ( Map.fromList
+            [ (c1, 12)
+            , (c2, 2)
+            , (c3, 7)
+            , (c4, 5)
+            ]
+        )
+    expectMembers Set.empty
+    expectNoFilterQueryResult Nothing
+    passEpoch >> passEpoch -- epoch 2
+    expectMembers $ Set.fromList [c1, c2, c3, c4]
+    -- members for which the expiration epoch is the current epoch are ToBeExpired
+    expectNoFilterQueryResult $
+      Just $
+        Map.fromList
+          [ (c1, CommitteeMemberState MemberNotAuthorized Active (Just 12) NoChangeExpected)
+          , (c2, CommitteeMemberState MemberNotAuthorized Active (Just 2) ToBeExpired)
+          , (c3, CommitteeMemberState MemberNotAuthorized Active (Just 7) NoChangeExpected)
+          , (c4, CommitteeMemberState MemberNotAuthorized Active (Just 5) NoChangeExpected)
+          ]
+    -- hot cred status of members with registered hot keys becomes MemberAuthorized
+    hk1 <- registerCommitteeHotKey c1
+    hk2 <- registerCommitteeHotKey c2
+    expectNoFilterQueryResult $
+      Just $
+        Map.fromList
+          [ (c1, CommitteeMemberState (MemberAuthorized (KeyHashObj hk1)) Active (Just 12) NoChangeExpected)
+          , (c2, CommitteeMemberState (MemberAuthorized (KeyHashObj hk2)) Active (Just 2) ToBeExpired)
+          , (c3, CommitteeMemberState MemberNotAuthorized Active (Just 7) NoChangeExpected)
+          , (c4, CommitteeMemberState MemberNotAuthorized Active (Just 5) NoChangeExpected)
+          ]
+    expectQueryResult
+      (Set.fromList [c2, c3, c5])
```

Comment:
Have you considered using `OverloadedLists` language extension in this module. It could reduce the noise quite a bit. This is just a suggestion.

Reply:
The extension was already declared in the file!  All i had to do is use it, thank you for making me aware of it. 
It removed quite a bit of noise (and tons of parentheses too  :skull: )